src/PoissonSolvers/LaplaceHelpers.h

```diff
@@ -0,0 +1,389 @@
+            meta_poisson(const E& u)
+                : u_m(u) {}
+
+            template <typename... Idx>
+            KOKKOS_INLINE_FUNCTION auto operator()(const Idx... args) const {
+                using index_type       = std::tuple_element_t<0, std::tuple<Idx...>>;
+                using T                = typename E::Mesh_t::value_type;
+                T res                  = 0;
+                index_type coords[dim] = {args...};
+                auto&& center          = apply(u_m, coords);
+                for (unsigned d = 0; d < dim; d++) {
+                    index_type coords[dim] = {args...};
+
+                    coords[d] -= 1;
+                    auto&& left = apply(u_m, coords);
+
+                    coords[d] += 2;
+                    auto&& right = apply(u_m, coords);
+                    res += (2.0 * center - left - right);
+                }
+                return res;
+            }
+
+        private:
+            const E u_m;
+        };
+
+        template <typename E>
+        struct meta_lower_laplace
+            : public Expression<meta_lower_laplace<E>,
+                                sizeof(E) + sizeof(typename E::Mesh_t::vector_type)
+                                    + 2 * sizeof(typename E::Layout_t::NDIndex_t)
+                                    + sizeof(unsigned)> {
+            constexpr static unsigned dim = E::dim;
+
+            KOKKOS_FUNCTION
+            meta_lower_laplace(const E& u, const typename E::Mesh_t::vector_type& hvector,
+                               unsigned nghosts, const typename E::Layout_t::NDIndex_t& ldom,
+                               const typename E::Layout_t::NDIndex_t& domain)
+                : u_m(u)
+                , hvector_m(hvector)
+                , nghosts_m(nghosts)
+                , ldom_m(ldom)
+                , domain_m(domain) {}
+
+            /*
+             * n-dimensional lower triangular Laplacian
+             */
+            template <typename... Idx>
+            KOKKOS_INLINE_FUNCTION auto operator()(const Idx... args) const {
+                using index_type = std::tuple_element_t<0, std::tuple<Idx...>>;
+                using T          = typename E::Mesh_t::value_type;
+                T res            = 0;
+
+                for (unsigned d = 0; d < dim; d++) {
+                    index_type coords[dim]       = {args...};
+                    const int global_index       = coords[d] + ldom_m[d].first() - nghosts_m;
+                    const int size               = domain_m.length()[d];
+                    const bool not_left_boundary = (global_index != 0);
+                    const bool right_boundary    = (global_index == size - 1);
+
+                    coords[d] -= 1;
+                    auto&& left = apply(u_m, coords);
+
+                    coords[d] += 2;
+                    auto&& right = apply(u_m, coords);
+
+                    res += hvector_m[d] * (not_left_boundary * left + right_boundary * right);
```

Comment:
maybe add a comment here to explain the not_left_boundary*left and the right_boundary*right